contrib_bots/lib/gcalendar.py

```diff
@@ -0,0 +1,234 @@
+    if not os.path.exists(credential_dir):
+        os.makedirs(credential_dir)
+    credential_path = os.path.join(credential_dir,
+                                   'gcalbot.json')
+
+    store = Storage(credential_path)
+    cred = store.get()
+    if not cred or cred.invalid:
+        flow = client.flow_from_clientsecrets(os.path.join(home_dir, CLIENT_SECRET_FILE), SCOPES)
+        flow.user_agent = APPLICATION_NAME
+        flags = argparser.parse_args('--noauth_local_webserver --logging_level INFO'.split())
+        cred = tools.run_flow(flow, store, flags)
+    return cred
+
+# This function currently has no use.
+# That's because right now there's no way of running the bot
+# with CLI options, which would be the best fit for this case.
+# The ideal scenario would be running the bot with the -l
+# option to see all available calendars, and execute it
+# (through a different option) with the desired calendar.
+#
+# Something like:
+#   $ ./run.py lib/gcalendar.py -l
+#       <list of calendars and ID's>
+#   $ ./run.py lib/gcalendar.py -c my_cal@group.calendar.google.com
+#       <the bot runs>
+def list_calendars():
+    '''Lists all the calendars with their corresponding ID's.
+
+    This is useful because calendar ID's are a bit of a hassle to
+    get, and are long, random strings that look like an email address.
+
+    The primary calendar is marked as "(primary)"
+    '''
+    cal_response = service.calendarList().list().execute()
+    calendar_list = cal_response.get('items', [])
+
+    for calendar in calendar_list:
+        name = calendar['summary']
+        cid = calendar['id']
+
+        if 'primary' in calendar:
+            name += ' (primary)'
+
+        print(name + ' - ID: ' + cid)
+
+def reply_message(msg, original_msg, client):
+    '''Replies to a specific message, through the channel of origin.
+
+    Keyword arguments:
+    msg -- content of the reply
+    original_msg -- message that is being replied
+    client -- client from the message handler
+    '''
+    message_type = original_msg['type']
+
+    if message_type == 'private':
+        original_sender = original_msg['sender_email']
+
+        client.send_message(dict(
+            type=message_type,
+            to=original_sender,
+            content=msg,
+        ))
+    elif message_type == 'stream':
+        original_stream = original_msg['display_recipient']
+        original_subject = original_msg['subject']
+
+        client.send_message(dict(
+            type=message_type,
+            to=original_stream,
+            subject=original_subject,
+            content=msg,
+        ))
+    else:
+        raise('Unknown message type')
+
+credentials = get_credentials()
+http = credentials.authorize(httplib2.Http())
+service = discovery.build('calendar', 'v3', http=http)
+
+class GCalendarHandler(object):
+    def __init__(self):
+        regex = ('(\w+|"[\S ]+") ((\d{4}-(0[1-9]|1[0-2])-(0[1-9]|1[0-9]|2[0-9]|3[0-1])'
```

Comment:
The already compiled regex is the one in a class level. I don't use the uncopiled one anywhere after compiling it, for the moment.

Is that what you mean?